gtdynamics/factors/PoseFactor.h

```diff
@@ -63,13 +61,13 @@ inline gtsam::ExpressionFactor<gtsam::Vector6> PoseFactor(const gtsam::SharedNoi
    * @param cost_model The noise model for this factor.
    * @param joint The joint connecting the two poses
    */
-inline gtsam::ExpressionFactor<gtsam::Vector6>  PoseFactor(DynamicsSymbol wTp_key, DynamicsSymbol wTc_key,
-             DynamicsSymbol q_key,
-             const gtsam::noiseModel::Base::shared_ptr &cost_model,
-             JointConstSharedPtr joint) {
-  return gtsam::ExpressionFactor<gtsam::Vector6>(
-             cost_model, gtsam::Vector6::Zero(),
-             joint->poseConstraint(wTp_key.time()));
+inline gtsam::NoiseModelFactor::shared_ptr PoseFactor(
```

Comment:
Is it the intention to return this as a NoiseModelFactor? Now the type info as an ExpressionFactor is lost so this may have repercussions that I am not sure of. @dellaert comments?

Reply:
One (debatable) downside I've noticed is that the test code now needs to use pointer syntax. So `factor.evaluateError` doesn't work and needs to be `factor->evaluateError`. Personally I am not a big fan because everywhere else, we use the former syntax and mixing-and-matching never goes well (aka greater cognitive overload aka less user friendly).

Reply:
These factors are on the way out. Let's stop spending time on them.

Reply:
And what will we be replacing them with? Or what is the expected use case? You can't simply make a blanket statement like that without considering the dependencies of others. :slightly_smiling_face:

Reply:
Varun, we have talked about this several times now: equalityConstints.If you’re not getting it yet, we can talk about it next Monday. For now, I want *all* those  factors gone from the wrapper unless there is code using them. AFAIK tell this is *only* you, formPoseFactor. correct me if I am wrong.